tezedge_state/src/proposer/proposal_persister.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) SimpleStaking, Viable Systems and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use std::fs::File;
+use std::io::prelude::*;
+use std::io::BufWriter;
+use std::sync::mpsc;
+
+use crate::proposals::*;
+
+pub struct ProposalPersisterHandle {
+    sender: Option<mpsc::SyncSender<RecordedProposal>>,
+    join_handle: Option<std::thread::JoinHandle<()>>,
+}
+
+impl Drop for ProposalPersisterHandle {
+    fn drop(&mut self) {
+        drop(self.sender.take());
+        if let Some(jh) = self.join_handle.take() {
+            jh.join().unwrap();
```

Comment:
@binier 
I am not sure, but I readsomewhere, that causing panic in drop, could lead to unexpected behavior,
and there could be possibly issue, that other Drops are not called or something like that,
I would rather do `match h.join()` and log or just do it without `unwrap`